.github/workflows/budibase_ci.yml

```diff
@@ -22,43 +22,44 @@ jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Use Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: 14.x
+          cache: "yarn"
       - run: yarn
       - run: yarn lint
 
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           submodules: true
           token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
       - name: Use Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: 14.x
+          cache: "yarn"
```

Comment:
Could you elaborate in the description what this will achieve for us? around what exactly is being cached / when does the cache invalidate

e.g. 
- When the pr first runs
- When the pr re-runs
- When the package.lock changes in subsequent commits 
etc.

Reply:
This does not cache any node_modules or similar, it caches the yarn cache. Every time that `yarn install` runs, the machine will download the packages and keep them locally (see `yarn cache dir`). Then these packages are then copied over to your node_modules, making subsequent `yarn install` much faster. This was the main selling point when `yarn` overtook `npm`, as it was much faster. You can test this by running `yarn cache clean` and rerunning yarn with a fresh cache. By default, GitHub action will run every time with a fresh cache. This is because it runs in dockerised instances.
Adding the cache key, these folders will be persisted between runs, pulling more from the local cache and less from the repos. https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data

The performance increase can be seen in the description message. You can see that the setup takes longer (as it needs to load the cache, but the yarn install is much quicker: 
<img width="829" alt="image" src="https://github.com/Budibase/budibase/assets/15987277/138a7a78-d6c7-466c-b2b7-adec4d3a0d3b">

Reply:
Thanks! Sounds like a great usage of the cache. So every run will start to benefit from this right away